plugin/my_asset_manager/AssetManagerInterface.py

```diff
@@ -0,0 +1,95 @@
+
+# Note that it should always be light-weight to construct instances of
+# the this class. See the notes under the "Initialization" section of:
+# https://openassetio.github.io/OpenAssetIO/classopenassetio_1_1v1_1_1manager_api_1_1_manager_interface.html#details (pylint: disable=line-too-long)
+# As such, any expensive module imports should be deferred.
+from openassetio import constants, BatchElementError, TraitsData
+from openassetio.managerApi import ManagerInterface
+
+# OpenAssetIO is building out the implementation vertically, there are
+# known fails for missing abstract methods.
+# pylint: disable=abstract-method
+# Methods in C++ end up with "missing docstring"
+# pylint: disable=missing-docstring
+# pylint: disable=too-many-arguments, unused-argument
+
+
+class MyAssetManagerInterface(ManagerInterface):
+    """
+    Implement the OpenAssetIO ManagerInterface.
+    https://openassetio.github.io/OpenAssetIO/classopenassetio_1_1manager_api_1_1_manager_interface_1_1_manager_interface.html
+    """
+
+    # Entity references provided to this asset manager should be
+    # prefixed with this string to be considered valid.
+    # eg. "my_asset_manager:///my_entity_id"
+    __reference_prefix = "my_asset_manager:///"
+
+    def identifier(self):
+        return "myorg.manager.my_asset_manager"
+
+    def initialize(self, managerSettings, hostSession):
+        pass
+
+    def displayName(self):
+        return "My Asset Manager"
+
+    def info(self):
+        # This hint allows the API middleware to short-circuit calls to
+        # `isEntityReferenceString` using string prefix comparisons. If
+        # your implementation's entity reference format supports this
+        # kind of matching, you should set this key. It allows for
+        # multi-threaded reference testing in C++ as it avoids the need
+        # to acquire the GIL and enter Python.
+        return {constants.kField_EntityReferencesMatchPrefix: self.__reference_prefix}
+
+    def managementPolicy(self, traitSets, context, hostSession):
+        # pylint: disable=unused-argument
+        return [TraitsData() for _ in traitSets]
+
+    def isEntityReferenceString(self, someString, hostSession):
+        return someString.startswith(self.__reference_prefix)
+
+    def resolve(
+        self, entityReferences, traitSet, context, hostSession, successCallback, errorCallback
+    ):
+        # Implement your resolution logic here.
+
+        # This is a dummy manager which only has the "anAsset" entry,
+        # and is not a good way to be doing things.
+        # For a better and more complete example implementation, which
+        # deals with things such as entity access modes and malformed
+        # entity references, see :
+        # https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/blob/main/plugin/openassetio_manager_bal/BasicAssetLibraryInterface.py
+        for idx, ref in enumerate(entityReferences):
+            if ref.toString() == "my_asset_manager:///anAsset":
```

Comment:
Not too sure about this. Wanted to be as direct as possible to avoid providing too many opinionated utilities in a template, but treating this as just a string is something i'm not sure about